### ISSUE TITLE
Pathfinder gives confident wrong answer with small sample prediction

### ISSUE THREAD

**Issue by fonnesbeck** [pymc-devs/pymc-experimental#279]
This example is taken from the baseball case study in `pymc-examples`.  We fit a beta-binomial model to some baseball batting data:

```python
data = pd.read_csv(pm.get_data("efron-morris-75-data.tsv"), sep="\t")

N = len(data)
player_names = data["FirstName"] + " " + data["LastName"]
# coords = {"player_names": player_names.tolist()}

with pm.Model() as baseball_model:
    at_bats = pm.MutableData("at_bats", data["At-Bats"].to_numpy())
    n_hits = pm.MutableData("n_hits", data["Hits"].to_numpy())
    baseball_model.add_coord("player_names", player_names, mutable=True)

    phi = pm.Uniform("phi", lower=0.0, upper=1.0)

    kappa_log = pm.Exponential("kappa_log", lam=1.5)
    kappa = pm.Deterministic("kappa", pm.math.exp(kappa_log))

    theta = pm.Beta("theta", alpha=phi * kappa, beta=(1.0 - phi) * kappa, dims="player_names")
    y = pm.Binomial("y", n=at_bats, p=theta, observed=n_hits, dims="player_names")
```

and then add a prediction for a fictional player that has zero hits in 4 appearances:

```python
with baseball_model:
    theta_new = pm.Beta("theta_new", alpha=phi * kappa, beta=(1.0 - phi) * kappa)
    y_new = pm.Binomial("y_new", n=4, p=theta_new, observed=0)
```

What should occur (and does with either `pymc.sample` or `pymc.fit`) is that since the sample size of `y_new` is so small, it should be shrunk towards the population mean. Here is the population of players:

![410871ea-80e5-44e6-a402-1c8acbd26ba6](https://github.com/pymc-devs/pymc-experimental/assets/81476/ff487f5b-88e9-4e47-ac4d-6ecb482eaf65)

and  the population mean is given by `phi`:

![112991f7-c433-4480-8026-e95aa154f926](https://github.com/pymc-devs/pymc-experimental/assets/81476/d0043fcc-b750-4493-bdbf-ce0a6c304c98)

however, the estimate for `theta_new` is way too large:

![adaf7323-a7f9-4a56-9a6b-1ae270ac6f0c](https://github.com/pymc-devs/pymc-experimental/assets/81476/f151be92-ea8f-4159-bb57-fad93035147d)

Using PyMC 3.10.1 and pymc-experimental from the main repo.

**Comment by junpenglao** [comment]
Not sure, the pathfinder return result that underestimated `kappa` and `theta`:
![image](https://github.com/pymc-devs/pymc-experimental/assets/12952641/d9f72c4c-9b58-4158-aa99-b86fc6715c58)

But probably this is the property of pathfinder, I dont work with it enough to provide good perspective. @ColCarroll has a bit more experience, maybe he has some idea?

**Comment by ricardoV94** [comment]
See also the issues I found before with the 8 school example, where it would return basically the initval for whatever mu was: https://gist.github.com/ricardoV94/eafd20ac47d63525253b0a8adf5e5d76

**Comment by junpenglao** [comment]
yeah the pathfinder have a jaxopt dependency that have some convergent gap (compare to [scipy.optimize.minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html)). I think on the blackjax side we can be more explicit on detecting none convergence.

**Comment by junpenglao** [comment]
For the intermediate, I suggest adding some noise to the initial position: https://github.com/pymc-devs/pymc-experimental/blob/00d7a2b3cf3379e0a9420fb436667ab781e5a5e7/pymc_experimental/inference/pathfinder.py#L104, so at the very least we can run the pathfinder a couple of times.

**Comment by ricardoV94** [comment]
Can you this to add jitter to RVs: https://github.com/pymc-devs/pymc/blob/0fd7b9e1d2208f1250b1c804bf5421013dba9023/pymc/initial_point.py#L111

**Comment by junpenglao** [comment]
Or both are wrong but in different places 😬
Is there any other implementation we can cross reference to?
